connectivity/lorawan/tests/UNITTESTS/features/lorawan/loraphycn779/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set(TEST_NAME lorawan-loraphy-cn779-unittest)
+
+add_executable(${TEST_NAME})
+
+target_compile_definitions(${TEST_NAME}
+    PRIVATE
+        MBED_CONF_LORA_DOWNLINK_PREAMBLE_LENGTH=5        
+        MBED_CONF_LORA_TX_MAX_SIZE=255
+        MBED_CONF_LORA_UPLINK_PREAMBLE_LENGTH=8
+)
+
+target_sources(${TEST_NAME}
+    PRIVATE
+        ${MBED_PATH}/connectivity/lorawan/lorastack/phy/LoRaPHYCN779.cpp
```

Comment:
it's a tradeoff, if you want to use library as dependency then you have to break up the module into tiny libraries since if you include the whole module then it will pull in too many dependencies, most of our modules are not self contained

Reply:
Yeah, it's a legit strategy to include just the implementation you are testing when unit testing. Listing the source file or two you are testing, rather than the library, give you an opportunity to replace that file's cpp or header dependencies with stubs, mocks, fakes, and so forth. We'd generally expect a unit test suite to include one real file, as is done here. If we are including 10s of real cpp files (we don't do that here, but do in other places in this PR), then that expresses a sort of need for a library, if not a need to rewrite the tests to be less integration or acceptance test and more unit test (i.e. rewrite the test to test a specific unit in isolation).

> most of our modules are not self contained

We should work on that separately.

For this PR, I suspect it is possible to avoid using `$MBED_PATH}`, even without having to link against a library target.

Reply:
IN this case if we want refer `LoRaPHYCN779.cpp` then we have to do `../../../../../` change directory it it better to refer with ${mbed-os_SOURCE_DIR}/connectivity/lorawan/lorastack/phy/LoRaPHYCN779.cpp" to give better visibility of actual source relevant path